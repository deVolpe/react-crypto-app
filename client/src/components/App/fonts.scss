@font-face {
  font: {
    family: 'Open Sans';
    weight: normal;
    style: normal;
  }
  src: url('./assets/fonts/OpenSans/OpenSans-Regular.eot');
  src: local('Open Sans Regular'), local('OpenSans-Regular'),
    url('./assets/fonts/OpenSans/OpenSans-Regular.eot')
      format('embedded-opentype'),
    url('./assets/fonts/OpenSans/OpenSans-Regular.woff') format('woff'),
    url('./assets/fonts/OpenSans/OpenSans-Regular.ttf') format('truetype');
}

@font-face {
  font: {
    family: 'Open Sans';
    weight: bold;
    style: normal;
  }
  src: url('./assets/fonts/OpenSans/OpenSans-Bold.eot');
  src: local('Open Sans Bold'), local('OpenSans-Bold'),
    url('./assets/fonts/OpenSans/OpenSans-Bold.eot?#iefix')
      format('embedded-opentype'),
    url('./assets/fonts/OpenSans/OpenSans-Bold.woff') format('woff'),
    url('./assets/fonts/OpenSans/OpenSans-Bold.ttf') format('truetype');
}
