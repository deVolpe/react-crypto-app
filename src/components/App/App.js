import React from 'react';
import PropTypes from 'prop-types';
import { BrowserRouter as Router, Route, Redirect } from 'react-router-dom';

import Header from '../../containers/HeaderContainer';
import Footer from '../Footer';
import Logo from '../Logo';
import MainSection from '../../containers/MainSectionContainer';
import Auth from '../Auth';
import Login from '../../containers/LoginContainer';
import Register from '../../containers/RegisterContainer';

import styles from './App.scss';

const App = ({ auth }) => (
  <Router>
    <div className={styles.App}>
      <Header>
        <Logo auth={auth} />
      </Header>
      <Route
        exact
        path="/"
        render={() => (auth.isAuthenticated ? (
          <Redirect to="/main/cards" />
        ) : (
          <Redirect to="/auth/login" />
        ))
          }
      />
      <Route
        path="/main"
        render={({ match }) => <MainSection match={match} />}
      />
      <Route
        path="/auth"
        render={({ match: { path } }) => (
          <Auth>
            <Route
              exact
              path={`${path}/login`}
              render={({ history: { push } }) => (
                <Login push={push} />
              )}
            />
            <Route
              exact
              path={`${path}/register`}
              render={({ history: { push } }) => (
                <Register push={push} />
              )}
            />
          </Auth>
        )}
      />
      <Footer>
        <Logo auth={auth} />
      </Footer>
    </div>
  </Router>
);

App.propTypes = {
  auth: PropTypes.shape({
    isAuthenticated: PropTypes.bool,
  }).isRequired,
};

export default App;
