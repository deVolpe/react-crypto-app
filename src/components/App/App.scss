.main {
  color: red;
  font-size: 15px;
  text-transform: uppercase;
}
