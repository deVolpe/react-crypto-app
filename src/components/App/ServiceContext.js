import { createContext } from 'react';

const {
  Provider: ServiceContextProvider,
  Consumer: ServiceContextConsumer,
} = createContext();

export { ServiceContextProvider, ServiceContextConsumer };
