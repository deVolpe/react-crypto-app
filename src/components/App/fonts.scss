@font-face {
  font: {
    family: 'Open Sans';
    weight: bold;
    style: normal;
  }
  src: url('./assets/fonts/Open_Sans/OpenSans-Bold.ttf') format('truetype');;
}

@font-face {
  font: {
    family: 'Open Sans';
    weight: normal;
    style: normal;
  }
  src: url('./assets/fonts/Open_Sans/OpenSans-Regular.ttf') format('truetype');;
}
