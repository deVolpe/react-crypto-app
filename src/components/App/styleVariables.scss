@import 'fonts';

$primary-font: 'Open Sans', sans-serif;
$background: #d1ceced9;
$style_background: #7b1ea7;
