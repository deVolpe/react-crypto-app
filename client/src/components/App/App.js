import React from 'react';
import PropTypes from 'prop-types';
import { BrowserRouter as Router, Route, Redirect } from 'react-router-dom';

import Header from '../../containers/HeaderContainer';
import Footer from '../Footer';
import Logo from '../Logo';
import Main from '../../containers/MainContainer';
import Auth from '../Auth';
import Login from '../../containers/LoginContainer';
import Register from '../../containers/RegisterContainer';
import Panel from '../Panel';
import Navbar from '../NavBar';
import SearchPanel from '../../containers/SearchPanelContainer';
import DropMenu from '../DropMenu';
import CardsList from '../../containers/CardsListContainer';
import { ServiceContextProvider } from './ServiceContext';
import CryptoCompareService from '../../services/cryptocompare-service';

import styles from './App.scss';

const App = ({ auth }) => {
  const service = new CryptoCompareService();
  return (
    <Router>
      <ServiceContextProvider value={service}>
        <div className={styles.App}>
          <Header>
            <Logo auth={auth} />
          </Header>
          <Route
            exact
            path="/"
            render={() => (auth.isAuthenticated ? (
              <Redirect to="/main/cards" />
            ) : (
              <Redirect to="/auth/login" />
            ))
            }
          />
          <Route
            path="/main"
            render={({ match: { path } }) => (
              <>
                <Panel>
                  <Navbar />
                  <SearchPanel />
                  <DropMenu />
                </Panel>
                <Main>
                  <Route
                    exact
                    path={`${path}/cards`}
                    render={() => <CardsList />}
                  />
                </Main>
              </>
            )}
          />
          <Route
            path="/auth"
            render={({ match: { path, url } }) => (
              <Auth>
                <Route
                  exact
                  path={`${path}/login`}
                  render={({ history: { push } }) => (
                    <Login push={push} url={url} />
                  )}
                />
                <Route
                  exact
                  path={`${path}/register`}
                  render={({ history: { push } }) => (
                    <Register push={push} url={url} />
                  )}
                />
              </Auth>
            )}
          />
          <Footer>
            <Logo auth={auth} />
          </Footer>
        </div>
      </ServiceContextProvider>
    </Router>
  );
};

App.propTypes = {
  auth: PropTypes.shape({
    isAuthenticated: PropTypes.bool,
  }).isRequired,
};

export default App;
