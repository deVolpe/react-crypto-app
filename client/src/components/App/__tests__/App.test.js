import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunkMiddleware from 'redux-thunk';

import App from '../../../containers/AppContainer';
// import store from '../../../modules/store';

const middlewares = [thunkMiddleware];
const mockStore = configureStore(middlewares);

describe('App', () => {
  test('should render auth=false', () => {
    const auth = { isAuthenticated: false };
    const wrapper = shallow(
      <Provider store={store}>
        <App auth={auth} />
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('[auth=true]', () => {
    const auth = { isAuthenticated: true };
    const wrapper = shallow(
      <Provider store={store}>
        <App auth={auth} />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
