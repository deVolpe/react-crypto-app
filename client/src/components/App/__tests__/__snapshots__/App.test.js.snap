// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App [auth=true] 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    }
  }
>
  <Connect(App)
    auth={
      Object {
        "isAuthenticated": true,
      }
    }
  />
</ContextProvider>
`;

exports[`App should render auth=false 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    }
  }
>
  <Connect(App)
    auth={
      Object {
        "isAuthenticated": false,
      }
    }
  />
</ContextProvider>
`;
